### PR TITLE
Basic Swift 6 Support

### PR DESCRIPTION
**Pull request by cobexer**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by cobexer** [comment]
@bot-gradle test this

**Comment by cobexer** [comment]
Swift 6 is not deployed on CI agents ATM, so I expect some failures, but an early review would not hurt IMO.

File: platforms/native/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/AbstractSwiftXCTestComponentWithTestedComponentIntegrationTest.groovy [hunk 1]
```diff
@@ -96,11 +97,12 @@ abstract class AbstractSwiftXCTestComponentWithTestedComponentIntegrationTest ex
         new Swift3WithSwift4XCTest('project') | SwiftVersion.SWIFT3          | SwiftVersion.SWIFT4
         new Swift4WithSwift3XCTest('project') | SwiftVersion.SWIFT4          | SwiftVersion.SWIFT3
         new Swift5WithSwift4XCTest('project') | SwiftVersion.SWIFT5          | SwiftVersion.SWIFT4
+        new Swift6WithSwift5XCTest('project') | SwiftVersion.SWIFT6          | SwiftVersion.SWIFT5
     }
 
     void assumeSwiftCompilerSupportsLanguageVersion(SwiftVersion swiftVersion) {
         assert toolChain != null, "You need to specify Swift tool chain requirement with 'requireSwiftToolChain()'"
-        assumeTrue((toolChain.version.major == 5 && swiftVersion.version in [5, 4]) || (toolChain.version.major == 4 && swiftVersion.version in [4, 3]) || (toolChain.version.major == 3 && swiftVersion.version == 3))
+        assumeTrue((toolChain.version.major == 6 && swiftVersion.version in [6, 5]) || (toolChain.version.major == 5 && swiftVersion.version in [5, 4]) || (toolChain.version.major == 4 && swiftVersion.version in [4, 3]) || (toolChain.version.major == 3 && swiftVersion.version == 3))
```
Comment (cobexer) [inline platforms/native/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/AbstractSwiftXCTestComponentWithTestedComponentIntegrationTest.groovy:105]: ❓ I could not find a source for this, so it's just an assumption at this point

File: testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json [hunk 2]
```diff
@@ -61,6 +61,14 @@
             "changes": [
                 "Method added to public class"
             ]
+        },
```
Comment (cobexer) [inline testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json:64]: ❓ This doesn't work, I did not have enough time to investigate why.

File: platforms/native/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/Swift6Test.groovy [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.nativeplatform.fixtures.app
+
+class Swift6Test extends XCTestSourceFileElement {
+    Swift6Test() {
+        super("Swift5Test")
```
Comment (ov7a) [inline platforms/native/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/Swift6Test.groovy:21]: 💅 should it be Swift6Test?

File: testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json (hunk 2 above)
Comment (cobexer) [reply testing/architecture-test/src/changes/accepted-changes/accepted-public-api-changes.json]: I changed the `changes` as well from the generated snipped ... and that is obviously wrong 🤦

**Comment by cobexer** [comment]
@bot-gradle test RfR

**Review by eskatos — APPROVED** [review]
On behalf of @bt-architecture-council

**Review by adammurdoch — APPROVED** [review]

**Comment by adammurdoch** [comment]
This is probably worth a mention in the release notes.

**Comment by cobexer** [comment]
@bot-gradle test RfR

**Comment by cobexer** [comment]
@adammurdoch could you take a look at this (and similar failures) https://ge.gradle.org/s/6mi6f5zqhw444/tests/task/:language-native:embeddedIntegTest/details/org.gradle.language.swift.SwiftApplicationIntegrationTest/can%20compile%20and%20link%20against%20a%20library%20with%20debug%20and%20release%20variants%20swiftc%206.0.1?top-execution=1

> Could not find source file 'greeter.swift' in symbols [SwiftRT-ELF-WASM.cpp, greeter.o]

I don't know anything about Swift and how it works with regards to debug symbols yet...


I think the cause of a large number of the other failures is:

```
error: emit-module command failed with exit code 1 (use -v to see invocation)	
<unknown>:0: error: implicit capture of 'self' requires that 'SwiftTest' conforms to `Sendable`	
/home/tcagent1/agent/work/f63322e10dd6b396/platforms/native/testing-native/build/tmp/test files/SwiftXCTest.Test/g30gj/build/linuxMain/main.swift:3:21: error: static property 'allTests' is not concurrency-safe because it is nonisolated global shared mutable state	
 1 | import XCTest	
 2 | extension SwiftTest {	
 3 |   public static var allTests = [	
   |                     |- error: static property 'allTests' is not concurrency-safe because it is nonisolated global shared mutable state	
   |                     |- note: convert 'allTests' to a 'let' constant to make 'Sendable' shared state immutable	
   |                     |- note: annotate 'allTests' with '@MainActor' if property should only be accessed from the main actor	
   |                     `- note: disable concurrency-safety checks if accesses are protected by an external synchronization mechanism	
 4 |     ("testSucceed", testSucceed),	
 5 |   ]	
/home/tcagent1/agent/work/f63322e10dd6b396/platforms/native/testing-native/build/tmp/test files/SwiftXCTest.Test/g30gj/src/test/swift/SwiftTest.swift:3:7: note: class 'SwiftTest' does not conform to the 'Sendable' protocol	
1 | import XCTest	
2 | 	
3 | class SwiftTest: XCTestCase {	
  |       `- note: class 'SwiftTest' does not conform to the 'Sendable' protocol	
4 |     func testSucceed() {	
5 |     XCTAssertTrue(true)
```

This sounds straight forward enough for me to make progress, but it's probably easier for you still =)

**Comment by ov7a** [comment]
Related:
* https://forums.swift.org/t/swift-5-10-concurrency-and-xctest/69929

**Comment by ov7a** [comment]
@bot-gradle test RfR

**Review by ov7a — APPROVED** [review]

**Pull request merged**